tests/test_spacecurve_casting.py

```diff
@@ -1,29 +1,23 @@
 import jax.numpy as jnp
 import numpy as np
 from qurveros.spacecurve import SpaceCurve
-
-def test_set_params_casts_int_to_float():
-    class DummyCurve(SpaceCurve):
+class DummyCurve(SpaceCurve):
         def __init__(self):
             super().__init__(curve=lambda x, p: jnp.array([x, x, x]), order=0, interval=[0, 1], params=None)
+
+def test_set_params_casts_int_to_float():
```

Comment:
This test needs to be inside a class for unit testing- see the existing tests in the `tests` folder for examples of how to refactor this properly.